pkg/state/test-ingress-state-missing-class.yaml

```diff
@@ -0,0 +1,122 @@
+#
+# OCI Native Ingress Controller
+#
+# Copyright (c) 2023 Oracle America, Inc. and its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
+#
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: ingress-state
+  namespace: default
+spec:
+  ingressClassName: default-ingress-class
+  tls:
+    - hosts:
+        - foo.bar.com
+      secretName: secret_name
+  rules:
+    - host: "foo.bar.com"
+      http:
+        paths:
+          - pathType: Prefix
+            path: "/PrefixEcho1"
+            backend:
+              service:
+                name: tls-test
+                port:
+                  number: 80
+    - host: "foo.bar.com"
+      http:
+        paths:
+          - pathType: Prefix
+            path: "/ExactEcho1"
+            backend:
+              service:
+                name: tls-test
+                port:
+                  number: 70
+---
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: ingress-state-new
+  namespace: default
+spec:
+  ingressClassName: default-ingress-class
+  tls:
+    - hosts:
+        - foo.bar.com
+      secretName: secret_name
+  rules:
+    - host: "foo.bar.com"
+      http:
+        paths:
+          - pathType: Prefix
+            path: "/PrefixEcho1/aa"
+            backend:
+              service:
+                name: tls-test
+                port:
+                  number: 80
+    - host: "foo.bar.com"
+      http:
+        paths:
+          - pathType: Prefix
+            path: "/ExactEcho1"
+            backend:
+              service:
+                name: tls-test
+                port:
+                  number: 90
+    - http:
+        paths:
+          - pathType: Prefix
+            path: "/PrefixEcho1"
+            backend:
+              service:
+                name: tls-test
+                port:
+                  number: 100
+---
+apiVersion: networking.k8s.io/v1
```

Comment:
Thinking this is a sequential processing, can we add ingress ingress-state-excluded in the middle and make sure the last ingress ingress-state-new is processed?